### DISCUSSION TITLE
Looking for timeline and more info on removing Java as a requirement for Batect

### DISCUSSION THREAD

**Discussion by salmanhoque-xero** [batect/batect#826] [General]
Hey 👋 ,

I really like Batect. I can easily see the benefit of this tools. It makes running task super consistent in any project. But currently Devs have to install Java. It would be great to see that requirement is removed and which will make Batect a self sufficient tool to run any task.

Luckily, in the [System requirements page](https://batect.dev/docs/getting-started/requirements) on the batect site I saw "_Java 8 or newer (**although this requirement will be removed before Batect v1.0**)_" . Really interested to know the timeline and the new implementation of Batect. Wondering is there any blog or docs I can follow? 

Thanks.

**Comment by charleskorn** [comment]
Hey @salmanhoque-xero, thanks for the question! Unfortunately I can't commit to any specific timeframe at this point - there are a couple of blockers that prevent transitioning the codebase from Kotlin/JVM (which requires Java to be installed when running the app) to Kotlin/Native (which would not require Java). These are things like the availability of a mocking library that works with Kotlin/Native, and some libraries that Batect depends on adding support for Kotlin/Native. I have seen a lot of activity in this area in the last 12 months, but there's still quite a bit left to go unfortunately. The good news is I'm really, really keen to do this, as it will improve the setup experience like you say, as well as offer a significant startup time improvement and make certain parts of the codebase much easier to maintain for me.

In the meantime, something I have been thinking about is making the first-run experience better if Java isn't installed. I've got a couple of ideas: either directing the user to a Java download page or even offering to automatically install Java for them (eg. through Homebrew on macOS or Chocolatey on Windows) if it's not installed already. Would one of these things help with your concerns? Or is the issue more of a gut 😱 reaction to the word Java by developers?

**Reply by salmanhoque-xero** [reply]
Hey @charleskorn sorry for the late reply, was stuck with a bunch of other things. 

Good to hear that it's still in the plan for Batect. I think this will make it a wonderful tool without having any requirement, since the tool itself uses other docker images for tasks.

I think in the meantime offering automatic Java installation might be a good step. This might help devs / juniors who are not familiar with various Java versions and things, less googling. 

I don’t mind java but always feels it’s gonna eat up all my memory. Good thing is, Batect just runs the task and not running a continuous process. :)
